src/main/java/com/xiao/users/service/impl/UserServiceImpl.java

```diff
@@ -48,4 +45,21 @@ public Page<UserDto> findAllUser(int pages, int pageSize) {
         return usersPage.map(userMapper::userToUserDto);
     }
 
+    @Override
+    public UserDto updateUser(Long userId, UserDto userDto){
+        userDto.setId(userId);
+        userRepository.findById(userId).orElseThrow(
+                () -> new ResourceNotFoundException("User", "id", String.valueOf(userId))
+        );
+
+        if(userDto.getUsername().isEmpty() && userDto.getEmailAddress().isEmpty() && userDto.getPassword().isEmpty()){
+            throw new EmptyAllFieldsUpdateException();
+        }
+
+        User userDataUpdate = userMapper.userDtoToUser(userDto);
```

Comment:
What happens if user only wants to update one field and doesn't provide values for other fields in the UserDto? Is our logic still correct?